docs/contributor/draft/rate-limit-design.md

```diff
@@ -0,0 +1,81 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| context | string | allows to set context for the underlying EnvoyFilter. Can be "gateway", which means local rate limit will be applied on the ingress Envoy proxies, or "sidecar" which means it will be applied on sidecar proxies of the given workload | yes                            |
+| workloadSelector                                  | object      | If omitted, the rate limit will be applied to all workload instances in the same namespace as the RateLimit CR. If the RateLimit CR is present in the istio-system namespace, it will be applied to all workloads in any namespace. | no                            |
+| workloadSelector.labels                          | map<string, string>      | One or more labels that indicate a specific set of pods/VMs on which the configuration should be applied. The scope of label search is restricted to the configuration namespace in which the the resource is present.                  | no                            |
+| local                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| local.bucket                            | object      | The token bucket configuration to use for rate limiting requests. Each request consumes a single token. If the token is available, the request will be allowed. If no tokens are available, the request will be rejected with status code 429.        | yes                           |
+| local.bucket.max_tokens                 | int         | The maximum tokens that the bucket can hold. This is also the number of tokens that the bucket initially contains.  | yes                           |
```

Comment:
I just realised that we have camel case and snake case mixed in the CR. We should only use camel case and rename `max_tokens`, `tokens_per_ fill`, ...

Reply:
Since IstioCR has all camelCase, I'd go with camelCase here. I think also the convention in k8s CRDs is to go with camelCase, if I'm not wrong.